File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        vector<int> w1(26, 0), w2(26, 0), w3(26, 0), w4(26, 0);
        for (char c : word1) {
            w1[c - 'a']++;
            w3[c - 'a'] = 1;
        }
        for (char c : word2) {
            w2[c - 'a']++;
            w4[c - 'a'] = 1;
        }
        sort(w1.begin(), w1.end());
        sort(w2.begin(), w2.end());
        return w1 == w2 && w3 == w4;
    }
};

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        int[] w1 = new int[26];
        int[] w2 = new int[26];
        boolean[] w3 = new boolean[26];
        boolean[] w4 = new boolean[26];
        for (int i = 0; i < word1.length(); i++) {
            char c = word1.charAt(i);
            w1[c - 'a']++;
            w3[c - 'a'] = true;
        }
        for (int i = 0; i < word2.length(); i++) {
            char c = word2.charAt(i);
            w2[c - 'a']++;
            w4[c - 'a'] = true;
        }
        Arrays.sort(w1);
        Arrays.sort(w2);
        for (int i = 0; i < 26; i++) {
            if (w1[i] != w2[i] || w3[i] != w4[i]) {
                return false;
            }
        }
        return true;
    }
}